### ISSUE TITLE
v3 - Fail fast for supplying schemas

### ISSUE THREAD

**Issue by good-idea** [prismicio/prismic-gatsby#109]
I just spent a *long* time trying to figure out some of my linked documents had the expected `document` field, while others did not. Turns out, it was because I had forgotten to add the schema to the plugin config.

I was eventually able to figure this out when I realized that the links that were working had a `PrismicLink` typename, while the ones that did not had an auto-generated one like `PrismicHomepageDataCategoriesCategory`.

Two things that could be handy:

 - Allow to simply pass the directory containing the schemas, instead of having to manually add them one by one.
 - If at all possible - throw an error if a schema is missing. This would ensure that us devs are supplying the schemas and taking advantage of all of the v3 benefits that come from it, and ensure that we get consistent results.

**Comment by good-idea** [comment]
Also, I'd like to say thanks for all of your work putting this together - it has made working with Prismic 100x easier.

**Comment by angeloashmore** [comment]
Thanks @good-idea, glad the improvements are helping.

This makes a lot of sense. The plugin will bail early if you don't provide any schemas, but it doesn't tell you if you are missing a schema.

If at some point we can get a list of all custom types present in the API data, we can check that against the list of provided schemas. This shouldn't be difficult to implement since the plugin grabs all documents and provides it as a large flat array.

**Comment by angeloashmore** [comment]
Re: providing a path to the schemas instead of an object, I prefer needing to pass them individually since it doesn't impose any file structure requirements on devs. By asking for a path to a folder containing all the schemas, it limits how files can be organized.

I've found this to be the easiest setup:

- Keep all schema JSON files in `src/schemas`.
- Create an `index.js` file that exports an object containing all the schemas. The exported object should match what the `schemas` plugin option is expecting.
- Provide the `index.js` object to the plugin options.

Something like:

```js
{
  options: {
    schemas: require('./src/schemas')
  }
}
```

This way `gatsby-config.js` no longer needs updating if a new schema if added. Of course, `src/schemas/index.js` still needs to be updated but I've found this to be easier to maintain.
